Control.cpp

```diff
@@ -34,60 +34,72 @@ double Control::getDesiredAngle(double desire) {
 double Control::getAngleError(const double current_angle_x, const double desire){
   double error = desire - current_angle_x;
   
-  if (error < -180) {
-     error += 360;
+  if (error < -kDegrees180) {
+     error += kDegrees360;
    }
     
-  else if (error > 180) {
-    error -= 360;
+  else if (error > kDegrees180) {
+    error -= kDegrees360;
   }
   
   return error;
 }
 
-double Control::getErrorUltrasonic(const double current_distance, const double desire) {
-  double error = current_distance - desire;
+double Control::getErrorUltrasonic(const double current_distance, const double desireUltrasonic) {
+  double error = current_distance - desireUltrasonic;
   return error;
 }
 
-void Control::getPwm(double &vel) { 
-  if (vel < kLimit_inf_pwm) {
-    vel = kLimit_inf_pwm;
+void Control::getPwm(double &speed) { 
+  if (speed < kLimit_inf_pwm) {
+    speed = kLimit_inf_pwm;
   }
   
-  else if (vel > kLimit_sup_pwm) {
-    vel = kLimit_sup_pwm;
+  else if (speed > kLimit_sup_pwm) {
+    speed = kLimit_sup_pwm;
   }
 }
 
 double Control::getNewDesireLeft(double new_desire) {
-  new_desire = new_desire - 90;
+  new_desire = new_desire - kDegrees90;
   if (new_desire < 0) {
-    new_desire = new_desire + 360;
+    new_desire = new_desire + kDegrees360;
   }
-  else {}
   return new_desire;
 }
 
 double Control::getNewDesireRight(double new_desire) {
-  new_desire = new_desire + 90;
-  if (new_desire > 360) {
-    new_desire = new_desire - 360;
+  new_desire = new_desire + kDegrees90;
+  if (new_desire > kDegrees360) {
+    new_desire = new_desire - kDegrees360;
   }
-  else{}
   return new_desire;
 }
 
 bool Control::detectRamp() {
-  if (logic.getAngleBNOY() < -4 && logic.getAngleBNOY() > -35) {
-    if (logic.getAngleBNOZ() < 3 && logic.getAngleBNOZ() > -3) {
-      return true;
-    }
-    else{
-      return false;
-    }
-  }
-  else{
-    return false;
-  }
+  return (bno_.getAngleY() < -kLimitInfDegrees 
+  && bno_.getAngleY() > -kLimitSupDegrees 
+  && bno_.getAngleZ() < kRangeAngleZ 
+  && bno_.getAngleZ() > -kRangeAngleZ);
+}
+
+bool Control::bumperLevel1() {
+  return (bno_.getAngleY() < -kLimitInfBumper1 
+  && bno_.getAngleY() > -kLimitSupBumper1 
+  && bno_.getAngleZ() > kRangeAngleZ 
+  || bno_.getAngleZ() < -kRangeAngleZ);
```

Comment:
Pay attention, the following two are not the same: 
(bno_.getAngleY() < -kLimitInfBumper1  && bno_.getAngleY() > -kLimitSupBumper1  && bno_.getAngleZ() > kRangeAngleZ  || bno_.getAngleZ() < -kRangeAngleZ)


(bno_.getAngleY() < -kLimitInfBumper1  && bno_.getAngleY() > -kLimitSupBumper1) && (bno_.getAngleZ() > kRangeAngleZ  || bno_.getAngleZ() < -kRangeAngleZ)